app/models/service_template_transformation_plan.rb

```diff
@@ -65,46 +66,20 @@ def self.create_catalog_item(options, _auth_user = nil)
     end
   end
 
-  private
-
-  def enforce_single_service_parent(_resource)
-  end
-
-  def self.validate_config_info(options)
-    config_info = options[:config_info]
-
-    mapping = if config_info[:transformation_mapping_id]
-                TransformationMapping.find(config_info[:transformation_mapping_id])
-              else
-                config_info[:transformation_mapping]
-              end
-
-    raise _('Must provide an existing transformation mapping') if mapping.blank?
-
-    pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
-    post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
+  def update_catalog_item(options, _auth_user = nil)
+    if miq_requests.any? || options[:config_info].nil?
```

Comment:
if `miq_requests` exists, do you allow an update?
If requests have been made and there were failed VMs, then user may want a retry. Before a retry they may want to update the VM list. Therefore it seems update should be allowed. But what about request is on-going? I think it is a question to PM.
Anyway, if it is not allowed, you should raise an error here. If allowed, you should use new `options`, not the existing one.

Reply:
Currently we are allowing Edits that includes Name/Description and VMs on only those plans that haven't been executed yet.
We won't be allowing editing the associated infrastructure mapping.

Edits won't be allowed for plans in-progress.

As far as failed and completed plans go, we would be supporting only partial edits that will let the user edit attributes like Name/Description

@mturley can you corroborate that?

Reply:
@AparnaKarve is correct, for failed and completed plans, the UI as designed only supports editing the name and description.

I do want to point out that although we currently don't allow editing the associated infrastructure mapping, that is mostly a UI restriction (it complicates the state of the wizard), and @vconzola mentioned we might want to allow that in the future (which basically would mean replacing an entire plan).

Reply:
@bzwei Addressed some of your above concerns in the last update (last commit)
Can you verify?

Reply:
To confirm when things can and cannot be edited, see this table: https://docs.google.com/document/d/1B-iFdNKqUgw44GPHc0TIUEDc9MdKiBll2pK6PTve__M/edit